doc/vxlan/EVPN/EVPN_VXLAN_HLD.md

```diff
@@ -0,0 +1,1352 @@
+ip link set dev vxlan-100 master Bridge
+bridge vlan add vid 100 dev vxlan-1000
+bridge vlan add vid 100 untagged pvid dev vxlan-100
+
+ip link set up vxlan-100
+```
+
+
+
+Note that in case of L3VNI, FRR requires VRF to be configured with corresponding L3VNI, and also L3VNI is bound to VRF in Linux. Below Linux operation is performed by intfmgr when interface is bound to VRF.
+
+```
+ip link set dev Vlan100 master <vrf>
+```
+
+vni netdev creation will trigger the IMET route generation. 
+
+Since this programming is dependent on VLAN to be present a check is being added as part of the CLI to ensure that VLAN is present before the map is configured.
+
+The existing kernel programming sequence in VxlanMgr for tunnels triggered by CFG_VNET_TABLE entries is not altered.
+
+##### FRR processing
+
+VLAN creation and a VLAN-VNI mapping in Linux triggers origination of IMET route in BGP.  
+FRR originates IMET routes based on kernel events listened to by zebra. 
+
+##### VxlanOrch processing
+
+In the current implementation, kernel programming for VXLAN devices is done by vxlanmgr only when CFG_VNET_TABLE entry is made. 
+
+In this enhancement for BGP EVPN tunnels, VXLAN_TUNNEL_MAP entry addition along with presence of VLAN will trigger the kernel programming. 
+
+
+### 4.3.7 MAC route handling
+
+
+MAC addresses learnt in the forwarding plane are populated by fdborch in the FDB_TABLE in STATE_DB. The Linux FDB learns only those MAC addresses which are CPU bound (e.g. from ARP request/response). In order to provide EVPN L2 service, all of the MAC addresses learnt in the forwarding plane need to be advertised.
+
+In order to achieve above, new process fdbsyncd is added in SwSS. Fdbsyncd would subscribe to FDB entries for a given VLAN once VLAN-VNI mapping is available. Fdbsyncd will maintain the VLAN subscription and send FDB events to Linux FDB.
+
+![EVPN MAC route handling](images/l2macflowcombined.PNG )
+__Figure 8: EVPN MAC route handling__
+
+
+Zebra will install all of the remote MAC routes to kernel. All Remote MAC addresses will be available in VXLAN_FDB_TABLE in APP_DB via Fdbsynd.
+
+```
+VXLAN_FDB_TABLE:{{vlan}}:{{mac_address}}
+    "remote_vtep" : {{ipv4_address}}
+    "type"   : "static" / "dynamic"  ; type of the entry.
+    "vni" : {{vni_id}}
+```
+
+In the case of MAC move from remote to local, remote MAC entry will be removed by Zebra from Linux FDB. Fdbsyncd will remove the remote mac from APP_VXLAN_FDB_TABLE when MAC moves.
+
+#### 4.3.7.1 Local MAC handling
+
+There is no change to fdborch wrt local MAC handling. 
+
+Currently Local MACs learnt by the fdborch are written to the STATE_DB. 
+
+Fdbsyncd listens to the above table and notifies Linux FDB which notifies to Zebra, Zebra to BGP and BGP originates the MAC route. 
+
+The above is applicable for locally added static MACs also. 
+
+![Local MAC Handling](images/localmac.PNG "Figure : Local MAC handling")
+__Figure 9: Local MAC handling__
```

Comment:
For the right side in the local mac handling flow chart, whether the fdbsyncd directly notify to zebra?? or it should write the fdb entry to kernel and then the zebra receive the kernel notification for the new local mac event

Reply:
Will update.